esp/platform/espp.hpp

```diff
@@ -182,14 +196,13 @@ class CEspServer : public CInterface,
         return applicationConfig.get();
     }
 
-    void log(LogLevel level, const char* fmt, ...) __attribute__((format(printf, 3, 4)))
+    void enableElevatedLogging()
     {
-        if (getLogLevel()>=level)
+        if (m_elevatedLoggingFilter == nullptr)
         {
-            va_list args;
-            va_start(args, fmt);
-            VALOG(MCdebugInfo, unknownJob, fmt, args);
-            va_end(args);
+            // Add custom log monitor to handle elevated context based logging
+            m_elevatedLoggingFilter = new CEspElevatedLoggingFilter();
```

Comment:
Does this need to be thread safe?

Reply:
Unlikely would be a problem. However, there is a small window where two requests could each be the first requesting elevated logging at the same time. @rpastrana Should we make this threadsafe?

Reply:
I would.

Reply:
yep, Ken's taking care of this per offline discussion

Reply:
If the user is setup for debug logging, this will currently be called a lot.. until caching of the MsgCatEspElevated is done.  You could consider optimizing so you rarely need the crit.

By checking for nullptr twice for example:
```
        if (m_elevatedLoggingFilter == nullptr)
        {
            CriticalBlock b(m_elevatedLoggingCs);
            if (m_elevatedLoggingFilter == nullptr)
            {
                // Add custom log monitor to handle elevated context based logging
                m_elevatedLoggingFilter = new CEspElevatedLoggingFilter();
                queryLogMsgManager()->addMonitorOwn(queryStderrLogMsgHandler(), m_elevatedLoggingFilter);
            }
        }
```